underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -313,14 +287,63 @@ protected ObjectStatus getObjectStatus(String key) {
       return new ObjectStatus(key, meta.getMd5HashAsBase64(), meta.getContentLength(),
           meta.getLastModifiedDate().getTime());
     } catch (ServiceException e) {
-      return null;
+      if (e.getResponseCode() == HttpStatus.SC_NOT_FOUND) {
+        return null;
+      }
+      throw new IOException(e);
     }
   }
 
-  // No group in GCS ACL, returns the account owner for group.
   @Override
   protected ObjectPermissions getPermissions() {
-    return new ObjectPermissions(mAccountOwner, mAccountOwner, mBucketMode);
+    return mPermissions.get();
+  }
+
+  /**
+   * Since there is no group in GCS acl, the owner is reused as the group. This method calls the
+   * GCS API and requires additional permissions aside from just read only. This method is best
+   * effort and will continue with default permissions (no owner, no group, 0700).
+   *
+   * @return the permissions associated with this under storage system
+   */
+  private ObjectPermissions getPermissionsInternal() {
+    // getAccountOwner() can return null even when the account is authenticated.
+    // TODO(chaomin): investigate the root cause why Google cloud service is returning
+    // null StorageOwner.
+    String accountOwnerId = "unknown";
+    String accountOwner = "unknown";
```

Comment:
These defaults are not consistent with the javadoc?

Reply:
Changed the default to `""` to consistent with javadoc